frontend/src/pages/admin_modals/index.tsx

```diff
@@ -169,6 +174,130 @@ export const AddUserModal = (props: {
   )
 }
 
+export const AddUserGroupModal = (props: {
+  onRequestClose: () => void,
+}) => {
+  const [isCompleted, setIsCompleted] = React.useState<boolean>(false)
+  const [includedUsers, setIncludedUsers] = React.useState(() => new Set());
```

Comment:
I think you should be able to modify this to remove the `as Set<string>` lines:

```suggestion
  const [includedUsers, setIncludedUsers] = React.useState<Set<string>>(() => new Set());
```
It's about the same as what you have though.